src/Simulation/decision_point.rs

```diff
@@ -0,0 +1,86 @@
+use itertools::Itertools;
+use regex::Regex;
+
+use crate::{
+    component::State,
+    TransitionSystems::{TransitionID, TransitionSystemPtr},
+};
+
+use super::transition_decision_point::TransitionDecisionPoint;
+
+/// Represents a decision in any composition of components: In the current `source` state there is a decision of using one of the `possible_decisions`.
+#[derive(Clone, Debug)]
+pub struct DecisionPoint {
+    source: State,
+    possible_decisions: Vec<String>,
+}
+
+impl DecisionPoint {
+    pub fn new(source: State, possible_decisions: Vec<String>) -> Self {
+        Self {
+            source,
+            possible_decisions,
+        }
+    }
+
+    pub fn source(&self) -> &State {
+        &self.source
+    }
+
+    pub fn possible_decisions(&self) -> &[String] {
+        self.possible_decisions.as_ref()
+    }
+
+    /// Returns the initial [`DecisionPoint`] in the given [`TransitionSystemPrt`].
+    pub fn initial(system: &TransitionSystemPtr) -> Option<Self> {
+        TransitionDecisionPoint::initial(system).map(|initial| DecisionPoint::from(&initial))
+    }
+}
+
+impl From<&TransitionDecisionPoint> for DecisionPoint {
+    fn from(transition_decision_point: &TransitionDecisionPoint) -> Self {
+        fn is_edge(x: &str) -> bool {
+            let is_not_edge_regex = Regex::new("(input_).*").unwrap(); // `.unwrap()` always return `Some(...)` here
+            !is_not_edge_regex.is_match(x)
+        }
+        let possible_decisions = transition_decision_point
+            .possible_decisions()
+            .iter()
+            .flat_map(|transition| transition.id.get_leaves().concat())
+            .filter_map(|transition_id| match transition_id {
+                TransitionID::Simple(v) => Some(v),
+                TransitionID::None => None,
+                _ => panic!("transition_id should not be other than Simple(_) and None"),
+            })
+            .unique()
+            .sorted()
```

Comment:
Do we need to sort here? If so, please sort after filter, so we don't have to sort as many elements.

Reply:
Yes, sorting here makes the API act deterministically in some cases, this makes testing a lot easier.